huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go

```diff
@@ -595,28 +615,15 @@ func resourceDdsInstanceV3Update(ctx context.Context, d *schema.ResourceData, me
 		opts = append(opts, opt)
 	}
 
-	if d.HasChange("port") {
-		resp, err := instances.UpdatePort(client, d.Id(), d.Get("port").(int))
-		if err != nil {
-			return diag.Errorf("error updating database access port: %s", err)
-		}
-		stateConf := &resource.StateChangeConf{
-			Pending:      []string{"Running"},
-			Target:       []string{"Completed"},
-			Refresh:      JobStateRefreshFunc(client, resp.JobId),
-			Timeout:      d.Timeout(schema.TimeoutUpdate),
-			PollInterval: 10 * time.Second,
-		}
-
-		_, err = stateConf.WaitForStateContext(ctx)
+	if len(opts) > 0 {
+		err := waitForInstanceReady(ctx, client, d.Id(), d.Timeout(schema.TimeoutUpdate))
```

Comment:
how about calling `waitForInstanceReady` at the beginning of `resourceDdsInstanceV3Update`?

Reply:
In an update, there may be multiple update operations performed

Reply:
updated